### PR TITLE
Support tokens, update for changed routes

### PR DESCRIPTION
**Pull request by fishnix**
A lot of changes necessary to support the new routes.  Moved from requiring ids to names for most space and resource operations.  Also support for token authentication and added a `configure` utility to write out a configuration file.

### CODE REVIEW

**Comment by damnski** [comment]
big change, to v3!  LGTM

**Review by FishNecks — APPROVED** [review]

File: cmd/configure.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	log "github.com/sirupsen/logrus"
+)
+
+func init() {
+	rootCmd.AddCommand(configureCmd)
+}
+
+var configureCmd = &cobra.Command{
+	Use:     "configure",
```
Comment (tenyo) [inline cmd/configure.go:18]: This is great 💯 so much easier to configure

File: pkg/spinup/server.go [hunk 2]
```diff
@@ -105,9 +105,11 @@ func (c *Client) ServerSize(id string) (*ServerSize, error) {
...
 		}
 
-		cpu = fmt.Sprintf("%0.00f vCPU", c/1024)
-		mem = fmt.Sprintf("%0.00f GB", m/1024)
+		size.CPU = fmt.Sprintf("%0.00f vCPU", c/1024)
+		size.Memory = fmt.Sprintf("%0.00f GB", m/1024)
 	}
 
-	return &ServerSize{size.(*BaseSize), cpu, mem}, nil
+	log.Debugf("returing server size %+v", size)
```
Comment (tenyo) [inline pkg/spinup/server.go:112]: 👀

**Review by tenyo — COMMENTED** [review]
Works great, just a few comments

File: README.md [hunk 3]
```diff
@@ -253,21 +249,20 @@ spinup list images -s 128 -s 136
         "volume_type": "gp2"
       }
     },
-    "offering_id": "18",
     "offering_name": "CentOS 7",
-    "space_id": "128"
+    "space_name": "funSpace"
```
[the remark this answers is not in the document]
Comment (fishnix) [reply README.md:253]: ah interesting - i think this needs some polish - lets work on it in the next pr - i think we decided to use the id on the backend calls but expose the name to the user.

File: pkg/spinup/resource.go [hunk 4]
```diff
@@ -1,26 +1,91 @@
...
+		req.Header.Set("Authorization", "Bearer "+c.AuthToken)
+	}
+
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed getting space "+space)
+	}
+
+	if res.StatusCode >= 400 {
+		return nil, errors.New("error getting space details: " + res.Status)
```
[the remark this answers is not in the document]
Comment (fishnix) [reply pkg/spinup/resource.go:63]: i think we should fix this in the UI - a 403 would make more sense in this case

**Pull request merged**